internal/clients/http/sources/sources_client.go

```diff
@@ -222,9 +222,15 @@ func (c *sourcesClient) GetAuthentication(ctx context.Context, sourceId string)
 	if resp == nil {
 		return nil, fmt.Errorf("get source authentication call: empty response: %w", clients.ErrUnexpectedBackendResponse)
 	}
-	if resp.JSON404 != nil || resp.JSON400 != nil {
+	if resp.JSON404 != nil {
 		logger.Warn().Bytes("body", resp.Body).Int("status", resp.StatusCode()).Msg("Get authentication from sources returned 4xx")
+		return nil, fmt.Errorf("get source authentication call: %w", clients.ErrNotFound)
 	}
+	if resp.JSON400 != nil {
+		logger.Warn().Bytes("body", resp.Body).Int("status", resp.StatusCode()).Msg("Get authentication from sources returned 4xx")
+		return nil, fmt.Errorf("get source authentication call: %w", clients.ErrBadRequest)
```

Comment:
The missing return at this resp.JSON404 != nil || resp.JSON400 != nil branch was the cause for the invalid returns.
The next **if resp.JSON200 == nil** returned ErrUnexpectedBackendResponse.
We are now validating id, so this would not happen for invalid ids at all. For 404 and 400 we are now able to return the correct status.